### PR TITLE
LPS-184538 Adapt patch and put methods for localized fields

### PR DESCRIPTION
**Pull request by seltong**
Follow-up: https://github.com/liferay-objects/liferay-portal/pull/2146

Related to: https://liferay.atlassian.net/browse/LPS-184538

### CODE REVIEW

**Comment by seltong** [comment]
@marcelabc hey, the last 2 commits are in the same file, but I did that in this case to help us decide what is the best approach for this case. I will talk to you about it on Slack

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java [hunk 1]
```diff
@@ -1674,6 +1673,27 @@ private void _deleteFromTable(
...
+
+		runSQL(
+			StringBundler.concat(
+				"delete from ",
+				dynamicObjectDefinitionLocalizationTable.getName(), " where ",
+				dynamicObjectDefinitionLocalizationTable.
+					getForeignKeyColumnName(),
+				" = ", objectEntryId));
+	}
+
```
Comment (marcelabc) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java:1701]: I saw that the method "_deleteFromTable()" has this code after running the SQL: `FinderCacheUtil.clearDSLQueryCache(dbTableName);`

I don't know if we should add it, just sharing with you.

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java [hunk 2]
```diff
@@ -2560,6 +2511,39 @@ else if (Objects.equals(
...
+						objectDefinitionId)) {
+
+				Map<String, String> localizedValues =
+					(Map<String, String>)expectedValues.get(
+						objectField.getI18nObjectFieldName());
+
+				String value = GetterUtil.get(
+					localizedValues.get(languageId), "");
+
+				expectedValues.put(objectField.getName(), value);
```
Comment (marcelabc) [inline modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java:2522]: I did not understand why we edit the expectedValues. Could you reach me and explain it, please? :smile:

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java [hunk 3]
```diff
@@ -1931,85 +1931,76 @@ public void testUpdateObjectEntry() throws Exception {
 			_objectDefinition2.getObjectDefinitionId(), objectEntry.getId(),
 			Arrays.asList("en_US", "fr_FR", "pt_BR"));
 
-		Map<String, String> localizedLongTextObjectFieldI18nValue =
```
Comment (marcelabc) [inline modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java]: This SF commit should be squashed with the previous commit; I left comments about a code that does not exist in the final version.

**Review by marcelabc — CHANGES_REQUESTED** [review]
Hi 
I finished my review. Let me know if you have questions :smile:

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/resource/v1_0/ObjectEntryResourceImpl.java [hunk 4]
```diff
@@ -474,6 +504,54 @@ protected String getPermissionCheckerResourceName(Object id)
...
+				(Map<String, String>)existingProperties.get(
+					objectField.getI18nObjectFieldName());
+
+			existingLocalizedValues.putAll(localizedValues);
+
+			existingProperties.put(
+				objectField.getI18nObjectFieldName(), existingLocalizedValues);
+		}
+
+		existingObjectEntry.setProperties(existingProperties);
```
[the remark this answers is not in the document]
Comment (seltong) [reply modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/resource/v1_0/ObjectEntryResourceImpl.java:566]: done

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/resource/v1_0/ObjectEntryResourceImpl.java [hunk 5]
```diff
@@ -266,6 +267,35 @@ aggregation, _getDTOConverterContext(null),
 			pagination, search, sorts);
 	}
 
+	@Override
+	public ObjectEntry patchObjectEntry(
```
[the remark this answers is not in the document]
Comment (seltong) [reply modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/resource/v1_0/ObjectEntryResourceImpl.java]: done

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/resource/v1_0/ObjectEntryResourceImpl.java [hunk 6]
```diff
@@ -474,6 +504,54 @@ protected String getPermissionCheckerResourceName(Object id)
...
+			}
+
+			Map<String, String> existingLocalizedValues =
+				(Map<String, String>)existingProperties.get(
+					objectField.getI18nObjectFieldName());
+
+			existingLocalizedValues.putAll(localizedValues);
+
+			existingProperties.put(
+				objectField.getI18nObjectFieldName(), existingLocalizedValues);
```
[the remark this answers is not in the document]
Comment (seltong) [reply modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/resource/v1_0/ObjectEntryResourceImpl.java:563]: done

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java [hunk 7]
```diff
@@ -1946,121 +1927,91 @@ public void testUpdateObjectEntry() throws Exception {
...
-						"localizedTextObjectFieldName_i18n",
-						HashMapBuilder.put(
-							"en_US", "en_US localizedTextObjectFieldValue"
-						).put(
-							"pt_BR", "pt_BR localizedTextObjectFieldValue"
-						).build()
-					).build();
-				}
-			});
+		Map<String, String> localizedRichTextObjectFieldI18nValue =
```
[the remark this answers is not in the document]
Comment (seltong) [reply modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java]: done

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java [hunk 8]
```diff
@@ -2560,6 +2511,39 @@ else if (Objects.equals(
 		}
 	}
 
+	private void _assertLocalizedValues(
+			Map<String, Object> expectedValues, long objectDefinitionId,
```
[the remark this answers is not in the document]
Comment (seltong) [reply modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java:2506]: done

**Comment by seltong** [comment]
ci:test:sf

**Comment by seltong** [comment]
ci:test:relevant

**Comment by seltong** [comment]
ci:test:relevant

**Review by marcelabc — APPROVED** [review]
LGTM 🎉

**Comment by seltong** [comment]
ci:test:sf

**Comment by seltong** [comment]
ci:test:object

**Comment by seltong** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by seltong** [comment]
ci:forward:force

**Pull request closed without merging**